mlflow/projects/__init__.py

```diff
@@ -280,6 +300,23 @@ def _fetch_project(uri, force_tempdir, version=None, git_username=None, git_pass
     return res
 
 
+def _unzip_repo(uri, dst_dir):
+    import zipfile
+    with zipfile.ZipFile(uri) as zip_file:
+        zip_file.extractall(dst_dir)
+
+
+def _fetch_zip_repo(uri, dst_dir):
+    import requests
+    import zipfile
+    from io import BytesIO
+    response = requests.get(uri)
+    if requests.codes.ok != response.status_code:
+        raise ExecutionException("Unable to retrieve ZIP file %s" % uri)
```

Comment:
Can we include the response code and, if possible, the response error text in the raised exception?